glprogs/bloom_downsample.frag.glsl

```diff
@@ -0,0 +1,41 @@
+#version 140
+#pragma tdm_define "BLOOM_BRIGHTPASS"
+
+/**
+ * This is the downsampling portion of the "dual filtering" blur as suggested in the Siggraph 2015 talk
+ * "Bandwidth-efficient Rendering" by Marius Bjorge.
+ */
+
+uniform sampler2D u_sourceTexture;
+uniform int u_sourceMipLevel;
+
+in vec2 var_TexCoord;
+out vec4 FragColor;
+
+#ifdef BLOOM_BRIGHTPASS
+const vec3 toGrayscale = vec3(0.2126, 0.7152, 0.0722);
+uniform float u_brightnessThreshold;
+uniform float u_thresholdFalloff;
+
+float brightpass(vec3 color) {
+	float brightness = dot(color.rgb, toGrayscale);
+	return clamp(pow(brightness / u_brightnessThreshold, u_thresholdFalloff), 0, 1);
+}
+#endif
+
+vec4 sampleTexture(vec2 offset) {
+	return textureLod(u_sourceTexture, var_TexCoord + offset, u_sourceMipLevel);
+}
+
+void main() {
+	vec2 halfpixel = 2 * vec2(0.5, 0.5) / textureSize(u_sourceTexture, u_sourceMipLevel);
```

Comment:
From a certain point of view, it is. But you're right, it may be more confusing than helpful, so I changed it to a more generic `offset` and added a short comment.